### ISSUE TITLE
twisted.mail.imap4 docs state incorrect meaning of UNSEEN value

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4271]
|<img alt="bobf's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| bobf reported|
|-|-|
|Trac ID|trac#4271|
|Type|defect|
|Created|2010-02-11 15:56:02Z|
|Branch|https://github.com/twisted/twisted/tree/imap4-unseen-docs-4271|

The docs for `IMAP4Client.select()` (http://twistedmatrix.com/documents/current/api/twisted.mail.imap4.IMAP4Client.html#select) state: "UNSEEN: An integer indicating the number of messages not flagged \Seen in this mailbox."

However, the RFC (http://www.faqs.org/rfcs/rfc3501.html) states:

"OK [UNSEEN <n>]
                     The message sequence number of the first unseen
                     message in the mailbox.  If this is missing, the
                     client can not make any assumptions about the first
                     unseen message in the mailbox, and needs to issue a
                     SEARCH command if it wants to find it."

The behaviour of IMAP4Client matches the RFC, not the Twisted docs (i.e. if I have one unread message, UNSEEN is the sequence number of that unread message, not "1").

<details><summary>Searchable metadata</summary>

```
trac-id__4271 4271
type__defect defect
reporter__bobf bobf
priority__normal normal
milestone__ 
branch__branches_imap4_unseen_docs_4271 branches/imap4-unseen-docs-4271
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1265903762000000 1265903762000000
changetime__1267635959000000 1267635959000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [28231]) Branching to 'imap4-unseen-docs-4271'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_3"></a>|@khorn set owner to @exarkun|
|-|-|

Looks good to me...merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [28643]) Merge imap4-unseen-docs-4271

Author: exarkun
Reviewer: khorn
Fixes: [#4271](https://github.com/twisted/twisted/issues/4271)

Correct the `twisted.mail.imap4` API documentation about the meaning of the
_UNSEEN_ flag in responses.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
